gh-109798: Normalize `_datetime` and `datetime` error messages (#127345)
python/cpython: Misc/NEWS.d/next/Library/2024-11-27-23-29-05.gh-issue-109798.OPj1CT.rst

Update error messages to be the same in :mod:`datetime`. Patch by Semyon

> Moroz.

Reviewer: Can you word the NEWS entry so it more clearly expresses the user visible change?

Author: @erlend-aasland I corrected the NEWS entry, but I'm not sure if the description is good enough, could you check?

> Normalized error messages for both implementations of :mod:`datetime`. Made
> more detailed error messages showing input values. Patch by Semyon Moroz.

Reviewer 2: I think we can shorten this. No need to explicitly mention that it was both implementations.
```suggestion
Normalized error messages for :mod:`datetime`, and made
them more useful when debugging. Patch by Semyon Moroz.
```
